main.go

```diff
@@ -82,11 +87,28 @@ func main() {
 
 	ctrl.SetLogger(zap.Logger(verboseLogging))
 
+	//When operator is started to watch resources in a specific set of namespaces, we use the MultiNamespacedCacheBuilder cache.
+	//In this scenario, it is also suggested to restrict the provided authorization to this namespace by replacing the default
+	//ClusterRole and ClusterRoleBinding to Role and RoleBinding respectively
+	//For further information see the kubernetes documentation about
+	//Using [RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/).
+	managerWatchCache := (cache.NewCacheFunc)(nil)
+	if namespaces != "" {
+		ns := strings.Split(namespaces, ",")
+		for i := range ns {
+			ns[i] = strings.TrimSpace(ns[i])
+		}
+		// always append empty namespace to watch cluster-scoped resources
+		// e.g. Nodes' labels are queried as part of broker rack-awareness setup
+		managerWatchCache = cache.MultiNamespacedCacheBuilder(append(ns, ""))
```

Comment:
If we append empty string wouldn't the operator watch all namespaces regardless of what has been passed in through the `--namespaces` flag?

Reply:
I'll double check that @stoader. My understanding is that empty ns in the controller cache stands for cluster scoped resources. But will double check

Reply:
Yes this can work. I'd also add to the note that of rack info is set manually than it's advisable to set pod affinity in such a way that the manually set rack info is in line with where the broker pod is scheduled to.

Reply:
Correct ^ - brokerGroup includes nodeAffinity to schedule the kafka pods in that group in particular failure-domain. Do you want me to add this in the docs here?

Reply:
yes please